crates/client/mempool/src/l1.rs

```diff
@@ -22,9 +24,19 @@ impl GasPriceProvider {
             data_gas_price_sync_enabled: Arc::new(AtomicBool::new(true)),
             strk_gas_price_sync_enabled: Arc::new(AtomicBool::new(true)),
             strk_data_gas_price_sync_enabled: Arc::new(AtomicBool::new(true)),
+            oracle_provider: None,
         }
     }
 
+    pub fn is_oracle_needed(&self) -> bool {
+        self.strk_gas_price_sync_enabled.load(Ordering::Relaxed)
+            || self.strk_data_gas_price_sync_enabled.load(Ordering::Relaxed)
+    }
+
+    pub fn set_oracle_provider(&mut self, oracle_provider: Oracle) {
+        self.oracle_provider = Some(Arc::new(oracle_provider));
+    }
```

Comment:
```suggestion
    pub fn set_oracle_provider(&mut self, oracle_provider: Oracle) -> &mut Self {
        self.oracle_provider = Some(Arc::new(oracle_provider));
        self
    }
```
allows method chaining by returning a reference to the current instance

Reply:
done